src/common/QueryHandler.ts

```diff
@@ -191,11 +210,60 @@ const pierceHandler = createInternalQueryHandler({
   },
 });
 
-const builtInHandlers = new Map([
-  ['aria', ariaHandler],
-  ['pierce', pierceHandler],
+const xpathHandler = internalizeCustomQueryHandler({
+  queryOne: (element, selector) => {
+    const doc = element.ownerDocument || document;
+    const result = doc.evaluate(
+      selector,
+      element,
+      null,
+      XPathResult.FIRST_ORDERED_NODE_TYPE
+    );
+    return result.singleNodeValue;
+  },
+
+  queryAll: (element, selector) => {
+    const doc = element.ownerDocument || document;
+    const iterator = doc.evaluate(
+      selector,
+      element,
+      null,
+      XPathResult.ORDERED_NODE_ITERATOR_TYPE
+    );
+    const array: Node[] = [];
+    let item;
+    while ((item = iterator.iterateNext())) {
+      array.push(item);
+    }
+    return array;
+  },
+});
+
+const transformXPathSelector = (selector: string): string => {
+  if (selector.startsWith('//')) {
+    selector = `.${selector}`;
+  }
+  return selector;
+};
+
+const INTERNAL_QUERY_HANDLERS = new Map<
+  string,
+  [
+    handler: InternalQueryHandler,
+    transformSelector?: (selector: string) => string
+  ]
+>([
+  ['aria', [ariaHandler]],
+  ['pierce', [pierceHandler]],
+  ['xpath', [xpathHandler, transformXPathSelector]],
```

Comment:
looks complicated? would it be better to move the transforms inside the xpath handler?

Reply:
The transforms update the selectors before handling. This is better since we aren't in page scope.

Reply:
hm, what about making a transformSelector to be part of the InternalQueryHandler API and providing an implementation for the xpathHandler?